core/pkg/service/flag-evaluation/connect_service.go

```diff
@@ -94,30 +95,39 @@ func (s *ConnectService) setupServer(svcConf service.Configuration) (net.Listene
 	path, handler := schemaConnectV1.NewServiceHandler(fes)
 	mux.Handle(path, handler)
 
-	mdlw := middleware.NewHTTPMetric(middleware.Config{
+	s.server = http.Server{
+		ReadHeaderTimeout: time.Second,
+		Handler:           handler,
+	}
+
+	go bindMetrics(s, svcConf)
```

Comment:
Since you are working on this (I am 100% agreeing to the refactoring)-

IMO this must be moved from `setupServer()` and be part of `serve()` - Because I expect services to be up and running when we actually serve :) 

See this implementation [1] for kube-proxy part which IMO is easy to understand 

And going further, we can avoid passing `ConnectService` to `bindMetrics()` as it's only usage is logging. I think we can pass a fresh logger for metrics which is even an improvement :) 

Do you think you can work on this too? Otherwise it could be a follow-up 

[1] - https://github.com/open-feature/flagd/blob/main/core/pkg/service/sync/server.go#L44-L45

Reply:
I agree with the first part @Kavindu-Dodan, great suggestion 👍  
With the suggestion for `bindMetrics()`, this function uses only the port field of the service configuration, not the logger. The function provides a binding for opening a fresh HTTP server for probes and the Prometheus metric page

Reply:
Good suggestion, I have moved the `bindMetrics` into the `Serve` method, and renamed it to `startMetricsServer` to be consistent with the `sync` server's naming. Another improvement would be to also move the logic of the start of the server handling the business logic into its own method, but I would do this in a small follow up PR after this one is merged :)

Reply:
Thank you :) look forward for the next PR